pkg/repair/repair.go

```diff
@@ -0,0 +1,220 @@
+// If the FieldSelectors struct is nil, it will create an empty one.
+func (f Filters) GetFieldSelectors() *KeyValueSelectorSet {
+	if f.FieldSelectors == nil {
+		f.FieldSelectors = &KeyValueSelectorSet{}
+	}
+	return f.FieldSelectors
+}
+
+// A safe getter for the LabelSelectors field of a Filters struct.
+// If the LabelSelectors struct is nil, it will create an empty one.
+func (f Filters) GetLabelSelectors() *KeyValueSelectorSet {
+	if f.LabelSelectors == nil {
+		f.LabelSelectors = &KeyValueSelectorSet{}
+	}
+	return f.LabelSelectors
+}
+
+// A struct wrapping a slice of field selector strings, used for convenience in
+// constructing a list of field selectors
+type KeyValueSelectorSet struct {
+	KeyValueSelectors []string
+}
+
+// Adds one or more selectors in format key=value to a KeyValueSelectorSet
+func (a *KeyValueSelectorSet) AddSelectors(selectors ...string) {
+	for _, selector := range selectors {
+		if selector != "" {
+			a.KeyValueSelectors = append(a.KeyValueSelectors, selector)
+		}
+	}
+}
+
+// Returns a stringified KeyValueSelectorSet
+func (a *KeyValueSelectorSet) String() string {
+	return strings.Join(a.KeyValueSelectors, ",")
+}
+
+// The pod reconciler struct. Contains state used to reconcile broken pods.
+type BrokenPodReconciler struct {
+	client  client.Interface
+	Filters *Filters
+	Options *Options
+}
+
+// Constructs a new BrokenPodReconciler struct.
+func NewBrokenPodReconciler(client client.Interface, filters *Filters, options *Options) (bpr BrokenPodReconciler) {
+	bpr = BrokenPodReconciler{
+		client:  client,
+		Filters: filters,
+		Options: options,
+	}
+	return
+}
+
+// Label all pods detected as broken by ListPods with a customizable label
+func (bpr BrokenPodReconciler) LabelBrokenPods() (err error) {
+	// Get a list of all broken pods
+	podList, err := bpr.ListBrokenPods()
+	if err != nil {
+		return
+	}
+
+	for _, pod := range podList.Items {
+		labels := pod.GetLabels()
+		if labels == nil {
+			labels = map[string]string{}
+		}
+		if _, ok := labels[bpr.Options.PodLabelKey]; ok {
+			log.Infof("Pod %s/%s already has label with key %s, skipping", pod.Namespace, pod.Name, bpr.Options.PodLabelKey)
+			continue
+		} else {
+			log.Infof("Labeling pod %s/%s with label %s=%s", pod.Namespace, pod.Name, bpr.Options.PodLabelKey, bpr.Options.PodLabelValue)
+			labels[bpr.Options.PodLabelKey] = bpr.Options.PodLabelValue
+			pod.SetLabels(labels)
+		}
+		if _, err = bpr.client.CoreV1().Pods(pod.Namespace).Update(&pod); err != nil {
+			return
+		}
+	}
+	return
+}
+
+// Delete all pods detected as broken by ListPods
+func (bpr BrokenPodReconciler) DeleteBrokenPods() error {
+	// Get a list of all broken pods
+	podList, err := bpr.ListBrokenPods()
+	if err != nil {
+		return err
+	}
+
+	for _, pod := range podList.Items {
+		log.Infof("Deleting broken pod: %s/%s", pod.Namespace, pod.Name)
+		if err := bpr.client.CoreV1().Pods(pod.Namespace).Delete(pod.Name, nil); err != nil {
+			return err
```

Comment:
Should not terminate loop on the first error. May return a list of errors.

Reply:
Fixed it by coalescing all the errors after the loop. Can you let me know if that isn't the approach you had in mind?